public/main.js

```diff
@@ -0,0 +1,49 @@
+function renderTip(i, tip) {
+  $button[i].setAttribute('data-tip', tip)
+}
+
+const $button = document.querySelectorAll('.button')
+const $topRight = document.querySelector('#top-right')
+const $topLeft = document.querySelector('#top-left')
+const $bottomright = document.querySelector('#bottom-right')
+const $bottomLeft = document.querySelector('#bottom-left')
+
+for (let i = 0; i < $button.length; i++) {
+  $button[i].addEventListener('mouseenter', function(event) {
+    const { top, bottom, left, right } = event.target.getBoundingClientRect()
+
+    console.log(`${top} px from the top of the window`)
+    console.log(
+      `${window.innerHeight - bottom}px from the bottom of the window`
+    )
+    console.log(`${left}px from the left of the window`)
+    console.log(`${window.innerWidth - right} from the right of the window`)
+    console.log(event.target)
+
+    if (
+      top < `${window.innerHeight - top}` &&
+      left < `${window.innerWidth - left}`
+    ) {
+      event.target.classList = 'button top-left'
```

Comment:
The Element.classList is a read-only property which returns a live DOMTokenList collection of the class attributes of the element.

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList